camayoc/tests/qcs/cli/test_scans.py

```diff
@@ -0,0 +1,332 @@
+    :param timeout: wait up to this amount of seconds. Default is 900.
+    """
+    while timeout > 0:
+        result = scan_show({'id': scan_id})
+        if status != 'failed' and result['status'] == 'failed':
+            raise FailedScanException(
+                'The scan with ID "{}" has failed unexpectedly.\n\n'
+                'The information about the scan is:\n{}\n'
+                .format(scan_id, pformat(result))
+            )
+        if result['status'] == status:
+            return
+        time.sleep(5)
+        timeout -= 5
+    raise WaitTimeError(
+        'Timeout waiting for scan with ID "{}" to achieve the "{}" status.\n\n'
+        'The information about the scan is:\n{}\n'
+        .format(scan_id, status, pformat(result))
+    )
+
+
+def test_scan(isolated_filesystem, qpc_server_config, source):
+    """Scan a single source type.
+
+    :id: 49ae6fef-ea41-4b91-b310-6054678bfbb4
+    :description: Perform a scan on a single source type.
+    :steps: Run ``qpc scan start --sources <source>``
+    :expectedresults: The scan must completed without any error and a report
+        should be available.
+    """
+    result = scan_start({
+        'source': source['name'],
+    })
+    match = re.match(r'Scan "(\d+)" started.', result)
+    assert match is not None
+    scan_id = match.group(1)
+    wait_for_scan(scan_id)
+    result = scan_show({
+        'id': scan_id,
+        'result': None,
+    })
+    assert 'connection_results' in result
+    assert 'inspection_results' in result
+
+
+def test_scan_with_multiple_sources(isolated_filesystem, qpc_server_config):
+    """Scan multiple source types.
+
+    :id: 58fde39c-52d8-42ee-af4c-1d75a6dc80b0
+    :description: Perform a scan on multiple source types.
+    :steps: Run ``qpc scan start --sources <source1> <source2> ...``
+    :expectedresults: The scan must completed without any error and a report
+        should be available.
+    """
+    result = scan_start({
+        'source': ' '.join([source['name'] for source in config_sources()]),
+    })
+    match = re.match(r'Scan "(\d+)" started.', result)
+    assert match is not None
+    scan_id = match.group(1)
+    wait_for_scan(scan_id, timeout=1200)
+    result = scan_show({
+        'id': scan_id,
+        'result': None,
+    })
+    assert 'connection_results' in result
+    assert 'inspection_results' in result
+
+
+def test_scan_with_disabled_products(isolated_filesystem, qpc_server_config):
+    """Perform a scan and disable an optional product.
+
+    :id: b1cd9901-44eb-4e71-846c-34e1b19751d0
+    :description: Perform a scan and disable an optional product.
+    :steps: Run ``qpc scan start --sources <source> --disable-optional-products
+        <optional-product>``
+    :expectedresults: The scan must completed without any error and a report
+        should be available. The disabled products should not have results in
+        the report.
+    :caseautomation: notautomated
+    """
```

Comment:
If it is not trouble could you add a similar test stub to the API tests with this test case?
I don't think I caught this scenario.

Reply:
I will prefer adding in a separated PR or a second commit to this. Better to have some logical separation.

Reply:
I'd appreciate it if you added a commit, I think it would be fewer total keystrokes to do that then write an issue to track needing to do that and making a separate PR.